src/geo/transform.js

```diff
@@ -844,20 +847,34 @@ class Transform {
             const dy = it.aabb.distanceY(cameraPoint);
             let dzSqr = cameraHeightSqr;
 
-            if (useElevationData) {
+            if (useElevationData || isGlobe) {
                 dzSqr = square(it.aabb.distanceZ(cameraPoint) * meterToTile);
             }
```

Comment:
You should double check if this logic is correct. Confusingly enough aabbs are defined in slightly different coordinate spaces depending on which projection is active. Originally the scale of the tile tree was chosen so that tiles on the target zoom level have size 1x1 with height (z) being defined in meters. This is still being used with all other projections but the globe which uses uniform scale for all axes.

If you look at how the `cameraPoint` is defined you'll notice that the elevation is in meters. For this reason reason the z-distance is multiplied with `meterToTile` before computing distance between the camera and the aabb. With globe projection aabbs are already in correct space and camera elevation should be updated too.